packages/react-instantsearch-hooks-dom/src/widgets/__tests__/index.test.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * @jest-environment node
+ */
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import {
+  InstantSearch,
+  InstantSearchServerContext,
+} from 'react-instantsearch-hooks';
+
+import * as allWidgets from '../';
+import { createSearchClient } from '../../../../../test/mock';
+
+import type { InstantSearch as InstantSearchClass } from 'instantsearch.js';
+
+/**
+ * Uses the SSR APIs to access the InstantSearch widgets rendered by all React InstantSearch
+ * components/widgets.
+ */
+function initializeWidgets() {
+  return Object.entries(allWidgets).map(([name, Component]) => {
+    let instantSearchInstance: InstantSearchClass | undefined = undefined;
+
+    renderToString(
+      <InstantSearchServerContext.Provider
+        value={{
+          notifyServer: ({ search }) => {
+            instantSearchInstance = search;
+          },
+        }}
+      >
+        <InstantSearch
+          searchClient={createSearchClient()}
+          indexName="indexName"
+        >
+          <Component />
+        </InstantSearch>
+      </InstantSearchServerContext.Provider>
+    );
+
+    const renderedWidgets = instantSearchInstance!.mainIndex.getWidgets();
+
+    return {
+      name,
+      renderedWidgets,
+      widget: renderedWidgets[0],
+    };
+  });
+}
+
+describe('widgets', () => {
+  const widgets = initializeWidgets();
+  test('renders one widget', () => {
+    widgets.forEach(({ name, renderedWidgets }) => {
+      expect({ name, renderedWidgets }).toEqual({
+        name,
+        renderedWidgets: expect.objectContaining({
+          length: 1,
+        }),
+      });
+    });
+  });
+
+  describe('$$type', () => {
+    test('present in every widget', () => {
+      widgets.forEach(({ name, renderedWidgets }) => {
+        renderedWidgets.forEach((widget) =>
+          expect([name, widget.$$type]).toEqual([name, expect.any(String)])
+        );
+      });
+    });
+
+    test('starts with ais.', () => {
+      widgets.forEach(({ name, widget }) =>
+        expect([name, widget.$$type.substr(0, 4)]).toEqual([name, 'ais.'])
+      );
+    });
+  });
+
+  describe('$$widgetType', () => {
+    test('present in every widget', () => {
+      widgets.forEach(({ name, widget }) =>
+        expect([name, widget.$$widgetType]).toEqual([name, expect.any(String)])
+      );
+    });
+
+    test('starts with ais.', () => {
```

Comment:
Maybe instead of asserting the prefix, we can map all values into an array of string.

This way we test:
1. That every widget has it (no wholes in the array)
2. That the names are stable (necessary for metadata not to have them change)

Since imports are alphabetically sorted with ESLint, this will only need to be updated when we add a new widget.

wdyt?

Reply:
Maybe the existing tests + an extra one that asserts the list of types? I'm not sure when we add a new item we will remember they need to have these requirements